install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,157 @@
+[[upgrading-blue-green-deployments]]
+= Blue-Green Deployments
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+Blue-green deployments are a proven approach to reducing downtime caused while
+updating an environment. This is done by creating a parallel environment on
+which the new deployment can be installed. If a problem is detected and after
+the new deployment is verified, traffic can be switched over with the option to
+rollback.
+
+While blue-green is a valid strategy for deploying just about any software,
+there are always trade-offs to be made. Not all environments have the same
+uptime requirements or the resources to properly perform blue-green deployments.
+In an {product-title} environment the most suitable candidate for blue-green
+deployments are the Nodes. Uptime is most important for these workloads and the
+additional complexity of blue-green deployments can be justified. Often the main
+challenge is having the excess capacity to facilitate such an approach.
+
+ifdef::openshift-enterprise[]
+Another lesser challenge is that the administrator will have to temporarily
+share the entitlements between the blue-green deployments or provide access to
+the installation content by means of a system such as Red Hat Satellite. This
+can be accomplished by sharing the consumer ID from the previous host.
+
+On the old host:
+----
+# subscription-manager identity | grep system
+system identity: 6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+On the new host:
+----
+# subscription-manager register --consumerid=6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+[IMPORTANT]
+====
+After a successful deployment, remember to unregister the old host with
+`*subscription-manager clean*` to prevent the environment from being out of
+compliance.
+====
+
+endif::openshift-enterprise[]
+
+[[manual-upgrades-preparing-for-upgrade]]
+== Preparing for Upgrade
+
+xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[After
+the master and etcd servers have been upgraded], you need to ensure that your
+current production nodes are labeled either blue or green. In this example, the
+current installation will be blue and the new environment will be green.
+
+----
+$ oc label --all nodes color=blue
+----
+
+In the case of nodes requiring the uptime guarantees of a blue-green deployment,
+the `-l` flag can be used to match a subset of the environment using a selector.
+
+Next, xref:../install/advanced_install.adoc#adding-nodes-advanced[the green
```

Comment:
Just FYI, @bfallonf has a PR (https://github.com/openshift/openshift-docs/pull/2570) that is proposing to move the "adding hosts" content to its own new topic. So that would affect this.